# HG changeset patch
# Parent 1f594a472590c038c27a1daf2e189e03e878691b
imported patch 3012-mips-lantiq-cpe-common-Kconfig-Makefile.patch

diff --git a/arch/mips/Kconfig b/arch/mips/Kconfig
old mode 100644
new mode 100755
--- a/arch/mips/Kconfig
+++ b/arch/mips/Kconfig
@@ -247,7 +247,7 @@ config LANTIQ
 	select SYS_SUPPORTS_MULTITHREADING
 	select SYS_HAS_EARLY_PRINTK
 	select ARCH_REQUIRE_GPIOLIB
-	select SWAP_IO_SPACE
+##	select SWAP_IO_SPACE
 	select BOOT_RAW
 	select HAVE_MACH_CLKDEV
 	select CLKDEV_LOOKUP
diff --git a/arch/mips/lantiq/Kconfig b/arch/mips/lantiq/Kconfig
old mode 100644
new mode 100755
--- a/arch/mips/lantiq/Kconfig
+++ b/arch/mips/lantiq/Kconfig
@@ -34,6 +34,15 @@ config DT_EASY50712
 	depends on SOC_XWAY
 endchoice
 
+config PCI_LANTIQ_BOOT
+	bool "PCI Boot Support"
+	depends on SOC_XWAY && PCI && !PCI_LANTIQ
+	default n
+	---help---
+	PCI bonding boot used for DSL off-chip bonding. Only xRX200 platform is
+	supported To use this option, xRX200 family board and bonding daughter
+	board must connect each other.
+
 config PCI_LANTIQ
 	bool "PCI Support"
 	depends on SOC_XWAY && PCI
@@ -41,6 +50,115 @@ config PCI_LANTIQ
 config PCIE_LANTIQ
 	bool "PCIE Support"
 	depends on SOC_XWAY && PCI
+choice
+	prompt "LANTIQ PCIE: Default PCIe PHY Clock Mode"
+	depends on PCIE_LANTIQ
+	default LANTIQ_PCIE_PHY_36MHZ_MODE
+	---help---
+	  Select ifx pci express phy clock mode by default.
+	config LANTIQ_PCIE_PHY_36MHZ_MODE
+	    bool "36MHz"
+
+	config LANTIQ_PCIE_PHY_36MHZ_SSC_MODE
+	    bool "36MHz SSC mode"
+
+	config LANTIQ_PCIE_PHY_25MHZ_MODE
+	    bool "25MHz"
+
+	config LANTIQ_PCIE_PHY_100MHZ_MODE
+	    bool "100MHz"
+endchoice
+
+config LANTIQ_PCIE_1ST_CORE
+	bool "Lantiq PCIE Core 0 Support"
+	depends on PCIE_LANTIQ
+	default y
+
+config LANTIQ_PCIE_HW_SWAP
+	bool "Lantiq PCIE Core 0 Outbound Hardware Swap Support"
+	depends on LANTIQ_PCIE_1ST_CORE
+	default y
+	---help---
+	  This will enable PCIe Core 0 Outbound hardware swap. It should be
+	  enabled together with PCI
+
+config LANTIQ_PCIE_INBOUND_NO_HW_SWAP
+	bool "Lantiq PCIE Core 0 Inbound Hardware Swap disabled"
+	depends on LANTIQ_PCIE_1ST_CORE
+	default n
+	---help---
+	  This will disable PCIe Core 0 Inbound Hardware Swap. It should be
+	  only used together with VRX318
+
+config LANTIQ_PCIE_RST_EP_ACTIVE_HIGH
+	bool "Lantiq PCIE Core 0 EP reset signal active high"
+	depends on LANTIQ_PCIE_1ST_CORE
+	default n
+	---help---
+	  Normally PCIe EP reset is low active. It should be only used
+	  together with VRX318 due to dying gasp
+
+config LANTIQ_PCIE_2ND_CORE
+	bool "Lantiq PCIE Core 1 Support"
+	depends on PCIE_LANTIQ
+	default n
+	---help---
+	  This will enable 2nd PCIe RC support
+
+config LANTIQ_PCIE1_HW_SWAP
+	bool "Lantiq PCIE Core 1 Outbound Hardware Swap Support"
+	depends on LANTIQ_PCIE_2ND_CORE
+	default n
+	---help---
+	  This will enable PCIe Core 1 Outbound data swap. It should be
+	  enabled together with PCI
+
+config LANTIQ_PCIE1_INBOUND_NO_HW_SWAP
+	bool "Lantiq PCIE Core 1 Inbound Hardware Swap Disabled"
+	depends on LANTIQ_PCIE_2ND_CORE
+	default n
+	---help---
+	  This will disable PCIe Core 1 Inbound Hardware Swap. It should be
+	  only used together with VRX318
+
+config LANTIQ_PCIE1_RST_EP_ACTIVE_HIGH
+	bool "Lantiq PCIE Core 1 EP reset signal active high"
+	depends on LANTIQ_PCIE_2ND_CORE
+	default n
+	---help---
+	  Normally PCIe EP reset is low active. It should be only used together
+	  with VRX318 due to dyning gasp
+
+config LANTIQ_PCIE_3RD_CORE
+	bool "Lantiq PCIE Core 2 Support"
+	depends on PCIE_LANTIQ
+	default n
+	---help---
+	  This will enable 3rd PCIe RC support
+
+config LANTIQ_PCIE2_HW_SWAP
+	bool "Lantiq PCIE Core 2 Outbound Hardware Swap Support"
+	depends on LANTIQ_PCIE_3RD_CORE
+	default n
+	---help---
+	  This will enable PCIe Core 2 Outbound data swap. It should be enabled
+	  together with PCI
+
+config LANTIQ_PCIE2_INBOUND_NO_HW_SWAP
+	bool "Lantiq PCIE Core 2 Inbound Hardware Swap Disabled"
+	depends on LANTIQ_PCIE_3RD_CORE
+	default n
+	---help---
+	  This will disable PCIe Core 2 Inbound Hardware Swap. It should be
+	  only used together with VRX318
+
+config LANTIQ_PCIE2_RST_EP_ACTIVE_HIGH
+	bool "Lantiq PCIE Core 2 EP reset signal active high"
+	depends on LANTIQ_PCIE_3RD_CORE
+	default n
+	---help---
+	  Normally PCIe EP reset is low active. It should be only used together
+	  with VRX318 due to dyning gasp
 
 config PCIE_LANTIQ_MSI
 	bool
diff --git a/arch/mips/lantiq/xway/Makefile b/arch/mips/lantiq/xway/Makefile
--- a/arch/mips/lantiq/xway/Makefile
+++ b/arch/mips/lantiq/xway/Makefile
@@ -1,4 +1,4 @@
-obj-y := prom.o sysctrl.o clk.o reset.o dma.o timer.o
+obj-y := prom.o sysctrl.o clk.o reset.o ltq_dma.o timer.o
 
 obj-y += vmmc.o mtd_split.o
 
